actions/get_single_sites.php

```diff
@@ -0,0 +1,162 @@
+<?php
+require_once "../../config.php";
+include '../tool-config.php';
+
+require_once("../dao/MigrateDAO.php");
+
+use \Tsugi\Core\LTIX;
+use \Migration\DAO\MigrateDAO;
+
+// Retrieve the launch data if present
+$LAUNCH = LTIX::requireData();
+
+$site_id = $LAUNCH->ltiRawParameter('context_id','none');
+
+$migrationDAO = new MigrateDAO($PDOX, $CFG->dbprefix);
+
+
+$result = ['success' => 0, 'msg' => 'requires POST'];
+
+$records_per_page = 10;  
+$output = '';
+
+if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+
+    if (isset($_POST['page'])) {
+        $page = $_POST["page"];  
+    } else {
+        $page = 1;  
+    }
+
+    if (isset($_POST['state'])) {
+        $state = $_POST['state'];
+    } else {
+        $state = "all";
+    }
+} else  {  
+    $page = 1;  
+}  
+
+
+$offset = ($page - 1) * $records_per_page;
+$result = $migrationDAO->getSingleSitesByState($LINK->id, $state, $offset, $records_per_page);
+
+$output .= "<table class='table flex-wrap' id='singlesites_tbl'><tbody>";  
+
+foreach($result as $row) {
+    $output .= '<tr>  
```

Comment:
You can probably simplify this row display by using a template (similar to the body / footer)

Reply:
created body and footer templates for single sites.